### PR TITLE
Added `pingpong` built-in function

### PR DESCRIPTION
**Pull request by Chaosus**
This math function used in Unity(https://docs.unity3d.com/ScriptReference/Mathf.PingPong.html), Blender, and (perhaps) many other engines. The proposal is to give users the same capability to use it as in them. This implementation is a copy from Blender with just cosmetical changes(should work as a Unity version, except for negative length values(which are not supported in the Unity for this function anyway)). Added to GDScript / C# / VisualScript.

### CODE REVIEW

**Comment by Calinou** [comment]
What are some practical use cases for this function? It's usually better to use `sin()` or `cos()` for smoothly changing pulsing lights. (Linear pulsing lights often look really bad.)

File: doc/classes/@GlobalScope.xml [hunk 1]
```diff
@@ -641,6 +641,29 @@
...
 		</method>
+		<method name="pingpong">
+			<return type="float">
+			</return>
+			<argument index="0" name="value" type="float">
+			</argument>
+			<argument index="1" name="length" type="float">
+			</argument>
+			<description>
+				Returns the [code]value[/code] wrapped between [code]0[/code] and the [code]length[/code]. If the limit is reached the next value the function returned is decreased to the [code]0[/code] side or increased to the [code]length[/code] side (like a wave). If [code]length[/code] is less than zero it become positive.
```
Comment (Calinou) [inline doc/classes/@GlobalScope.xml:652]:
```suggestion
				Returns the [code]value[/code] wrapped between [code]0[/code] and the [code]length[/code]. If the limit is reached, the next value the function returned is decreased to the [code]0[/code] side or increased to the [code]length[/code] side (like a triangle wave). If [code]length[/code] is less than zero, it becomes positive.
```

**Comment by Chaosus** [comment]
@Calinou I have no idea by myself since I'm not using it, maybe some former Unity users may say why they like it and whether they want it in the Godot.

File: doc/classes/@GlobalScope.xml (hunk 1 above)
Comment (Chaosus) [reply doc/classes/@GlobalScope.xml]: Fixed, I guess...

**Comment by aaronfranke** [comment]
@Chaosus We don't just add features "just because". This needs a proposal. If you aren't using it and don't know how it will be used, that is hardly a substantive use case. So I'm closing this as per the Godot proposal policy (since there is no use case here, one needs to be presented in a proposal before this can be considered).

> 3. All proposals must be linked to a substantive use-case. In justifying your proposal, it is not enough to say it would be "nice" or "helpful". Use the template to show how Godot is not currently meeting your needs and then explain how your proposal will meet a particular need.

**Comment by Chaosus** [comment]
Ok, I forgot about it a bit...

**Comment by TokageItLab** [comment]
Please restore this PR, so I need this for #48332!

That PR is to implement a ping-pong loop in the animation function, and without this function, you will have to write the same mathematical expression again and again.

In short, instead of normal loop using `fposmod()`, the ping-pong loop needs `pingpong()`.

**Comment by fire** [comment]
@Chaosus is currently unavailable, but we we discussed that if we are able to find a use, we can use it.

So we can put `pingpong` into your pr.

**Comment by TokageItLab** [comment]
Yes, it has been merged into my PR, so there is no need to necessarily restore this one unless a step-by-step merge is required.

**Comment by aaronfranke** [comment]
For the record, this has been implemented in https://github.com/godotengine/godot/pull/53819

**Pull request closed without merging**